widgets/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/validators/RunOnceEditorValidator.java

```diff
@@ -35,17 +36,41 @@ public boolean isValid() {
     if ( null == editor.getStartDate() ) {
       isValid = false;
     } else {
+
       final DateTimeFormat format = DateTimeFormat.getFormat( "MM-dd-yyyy" ); //$NON-NLS-1$
       final String date = format.format( editor.getStartDate() );
-      final String dateTime = date + " " + editor.getStartTime(); //$NON-NLS-1$
+      String time = editor.getStartTime();  //format of time is "hh:mm:ss a"
+
+      if (editor.getStartDate().before( new Date() )){
+       isValid=false;
+      } else {
+
+        if(time.endsWith( TimeUtil.TimeOfDay.AM.toString() ) || time.endsWith( (TimeUtil.TimeOfDay.PM.toString() ) )){
+          time = normalizeTime( time );
+        }
+
+        final String dateTime = date + " " + time; //$NON-NLS-1$
+
+        if ( DateTimeFormat.getFormat( "MM-dd-yyyy hh:mm:ss a" ).parse( dateTime ).before( new Date() ) ) { //$NON-NLS-1$
+          isValid = false;
+        }
 
-      if ( DateTimeFormat.getFormat( "MM-dd-yyyy hh:mm:ss a" ).parse( dateTime ).before( new Date() ) ) { //$NON-NLS-1$
-        isValid = false;
       }
     }
     return isValid;
   }
 
   public void clear() {
   }
+
+  private String normalizeTime( String time ) {
+    if ( time != null && !time.isEmpty() &&
+      ( time.endsWith( TimeUtil.TimeOfDay.AM.toString() ) || time.endsWith( TimeUtil.TimeOfDay.PM.toString() ) ) ) {
+      if ( time.endsWith( TimeUtil.TimeOfDay.AM.toString() ))
+        time = time.replace( TimeUtil.TimeOfDay.AM.toString(), "am");
+      else
+        time = time.replace( TimeUtil.TimeOfDay.PM.toString(), "pm");
+    }
+    return time;
```

Comment:
IMO, would be simpler if the AM/PM tests would only be done by the inner `if`/`else`, while transforming the `else` into an `else if` for PM.